onedal/linear_model/logistic_regression.py

```diff
@@ -0,0 +1,239 @@
+        params = self._get_onedal_params(get_dtype(X))
+        X_table, y_table = to_table(X, y)
+
+        result = module.train(policy, params, X_table, y_table)
+
+        self._onedal_model = result.model
+        self.n_iter_ = np.array([result.iterations_count])
+
+        coeff = from_table(result.model.packed_coefficients)
+        self.coef_, self.intercept_ = coeff[:, 1:], coeff[:, 0]
+
+        return self
+
+    def _create_model(self, module, policy):
+        m = module.model()
+
+        coefficients = self.coef_
+        dtype = get_dtype(coefficients)
+        coefficients = np.asarray(coefficients, dtype=dtype)
+
+        if coefficients.ndim == 2:
+            n_features_in = coefficients.shape[1]
+            assert coefficients.shape[0] == 1
+        else:
+            n_features_in = coefficients.size
+
+        intercept = self.intercept_
+        if not isinstance(intercept, Number):
+            intercept = np.asarray(intercept, dtype=dtype)
+            assert intercept.size == 1
+
+        intercept = _check_array(
+            intercept,
+            dtype=[np.float64, np.float32],
+            force_all_finite=True,
+            ensure_2d=False,
+        )
+        coefficients = _check_array(
+            coefficients,
+            dtype=[np.float64, np.float32],
+            force_all_finite=True,
+            ensure_2d=False,
+        )
+
+        coefficients, intercept = make2d(coefficients), make2d(intercept)
+        coefficients = coefficients.T
+
+        assert coefficients.shape == (1, n_features_in)
+        assert intercept.shape == (1, 1)
+
+        desired_shape = (1, n_features_in + 1)
+        packed_coefficients = np.zeros(desired_shape, dtype=dtype)
+
+        packed_coefficients[:, 1:] = coefficients
+        if self.fit_intercept:
+            packed_coefficients[:, 0][:, np.newaxis] = intercept
+
+        packed_coefficients = _convert_to_supported(policy, packed_coefficients)
+
+        m.packed_coefficients = to_table(packed_coefficients)
+
+        self._onedal_model = m
+
+        return m
+
+    def _infer(self, X, module, queue):
+        _check_is_fitted(self)
+
+        policy = self._get_policy(queue, X)
+
+        X = np.asarray(X)
+
+        # Finiteness is checked in the sklearnex wrapper
+        X = _check_array(
+            X, dtype=[np.float64, np.float32], force_all_finite=False, ensure_2d=False
+        )
+        _check_n_features(self, X, False)
+
+        if hasattr(self, "_onedal_model"):
+            model = self._onedal_model
+        else:
+            model = self._create_model(module, policy)
+
+        X = make2d(X)
+        X = _convert_to_supported(policy, X)
+        params = self._get_onedal_params(get_dtype(X))
+
+        X_table = to_table(X)
+        result = module.infer(policy, params, model, X_table)
+        return result
+
+    def _predict(self, X, module, queue):
+        result = self._infer(X, module, queue)
+        y = from_table(result.responses)
```

Comment:
It would seem to me that two functions on the cpp side would be better served than to have two attributes of a result object which are never used together (_predict doesn't use result.probabilities and _predict_proba doesnt use result.responses)

Reply:
I noticed that for now we cannot specify whether we can